### PR TITLE
pkg/updates: move logic into graph methods

### PR DESCRIPTION
**Pull request by jzelinskie**
This PR refactors a bit of the logic being used to traverse the UpdateGraph into methods specifically on the update graph and adds tests for them.

### CODE REVIEW

File: pkg/updates/file_test.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+			Metadata: map[string]string{"datastore": "cockroachdb"},
+			Nodes:    []State{{ID: "v1.0.0"}},
+		},
+	}}
+
+	t.Run("common case", func(t *testing.T) {
+		channel, err := graph.ChannelForDatastore("cockroachdb")
+		require.Nil(t, err)
+		require.Equal(t, "cockroachdb", channel)
+		channel, err = graph.ChannelForDatastore("postgres")
```
Comment (josephschorr) [inline pkg/updates/file_test.go:29]: newline before this line

File: pkg/updates/file_test.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+		{
+			name:          "empty graph",
+			graph:         &UpdateGraph{},
+			engine:        "postgres",
+			version:       v1alpha1.SpiceDBVersion{Name: "v1.0.0", Channel: "postgres"},
+			expectedNames: nil,
+			expectedErr:   `no source found for channel "postgres"`,
+		},
+		{
+			name: "graph without channel",
```
Comment (josephschorr) [inline pkg/updates/file_test.go:59]: without matching channel

File: pkg/updates/file_test.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+			name: "graph without edges",
+			graph: &UpdateGraph{Channels: []Channel{{
+				Name:     "cockroachdb",
+				Metadata: map[string]string{"datastore": "cockroachdb"},
+				Nodes:    []State{{ID: "v1.0.1"}, {ID: "v1.0.0"}},
+			}}},
+			engine:        "cockroachdb",
+			version:       v1alpha1.SpiceDBVersion{Name: "v1.0.0", Channel: "cockroachdb"},
+			expectedNames: nil,
+			expectedErr:   "no edges or no nodes",
```
Comment (josephschorr) [inline pkg/updates/file_test.go:80]: I wonder if we can make this error more descriptive

File: pkg/updates/file_test.go [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+			name: "simple patch update",
+			graph: &UpdateGraph{Channels: []Channel{{
+				Name:     "cockroachdb",
+				Metadata: map[string]string{"datastore": "cockroachdb"},
+				Edges:    EdgeSet{"v1.0.0": {"v1.0.1"}},
+				Nodes:    []State{{ID: "v1.0.1"}, {ID: "v1.0.0"}},
+			}}},
+			engine:        "cockroachdb",
+			version:       v1alpha1.SpiceDBVersion{Name: "v1.0.0", Channel: "cockroachdb"},
+			expectedNames: []string{"v1.0.1"},
```
Comment (josephschorr) [inline pkg/updates/file_test.go:104]: can you add a test starting a v1.0.1 too and make sure it does nothing?

File: pkg/updates/file_test.go (hunk 3 above)
Comment (jzelinskie) [reply pkg/updates/file_test.go:80]: The error is more descriptive, I'm only checking that this string is a substring of the error

Comment (josephschorr) [reply pkg/updates/file_test.go:81]: I meant more to only say "edges" or "nodes", if missing, rather than both

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**